### PR TITLE
Add deprecation notice

### PR DESCRIPTION
**Pull request by jorgensd**
- Add deprecation notices on unused functions from `common`.
- Restructure to avoid circular import
- Fix face_topology (should be facet topology)
- Add some typing

@emmetfrancis It would be good to know if you see any deprecation warnings when running this branch.
If you do, it means that you are using functions that are not called anywhere in stubs.
We should then add appropriate tests for them.

### CODE REVIEW

File: stubs/common.py [hunk 1]
```diff
@@ -186,7 +273,7 @@ def mesh_vertex_to_dof(V, species_index, index):
     return [mapping[x] for x in index]
 
 
-def round_to_n(x, n):
+def round_to_n(x: np.float64, n: int) -> int:
     """
```
Comment (finsberg) [inline stubs/common.py]: Do we need `np.float64` here or is `float` sufficient?

**Review by finsberg — APPROVED** [review]

**Comment by finsberg** [comment]
I see that the example notebook also has been changed. I think we should strip out the output from the notebooks before we commit them so that we avoid to many changes to these files. We can for example use: https://github.com/kynan/nbstripout which can be added as a pre-commit hook. See e.g https://github.com/finsberg/fenicsx-pulse/blob/main/.pre-commit-config.yaml#L49

**Comment by jorgensd** [comment]
> I see that the example notebook also has been changed. I think we should strip out the output from the notebooks before we commit them so that we avoid to many changes to these files. We can for example use: https://github.com/kynan/nbstripout which can be added as a pre-commit hook. See e.g https://github.com/finsberg/fenicsx-pulse/blob/main/.pre-commit-config.yaml#L49

For now I've just stripped the output manually.

**Comment by emmetfrancis** [comment]
> * Add deprecation notices on unused functions from `common`.
> * Restructure to avoid circular import
> * Fix face_topology (should be facet topology)
> * Add some typing
> 
> @emmetfrancis It would be good to know if you see any deprecation warnings when running this branch. If you do, it means that you are using functions that are not called anywhere in stubs. We should then add appropriate tests for them.

I believe the two functions (read_sbmodel and write_sbmodel) are not used in any of the current code I have from myself or Justin, but I will let you know if the warnings pop up. Thanks for the fixes here; I will be sending another pull request soon with refined versions of the other examples I have been working on.

**Pull request merged**